spyder/plugins/completion/languageserver/client.py

```diff
@@ -193,93 +202,82 @@ def start(self):
                 elif get_debug_level() == 3:
                     self.server_args.append('-vv')
 
-        server_stdin = subprocess.PIPE
-        server_stdout = server_log
-        server_stderr = subprocess.STDOUT
-
-        if not self.external_server:
-            logger.info('Starting server: {0}'.format(
-                ' '.join(self.server_args)))
-            creation_flags = 0
-            if os.name == 'nt':
-                creation_flags = (subprocess.CREATE_NEW_PROCESS_GROUP
-                                  | 0x08000000)  # CREATE_NO_WINDOW
-
-            if os.environ.get('CI') and os.name == 'nt':
-                # The following patching avoids:
-                #
-                # OSError: [WinError 6] The handle is invalid
-                #
-                # while running our tests in CI services on Windows
-                # (they run fine locally).
-                # See this comment for an explanation:
-                # https://stackoverflow.com/q/43966523/
-                # 438386#comment74964124_43966523
-                def patched_cleanup():
-                    pass
-                subprocess._cleanup = patched_cleanup
+        logger.info('Starting server: {0}'.format(' '.join(self.server_args)))
 
-            # Set the PyLS current working to an empty dir inside
-            # our config one. This avoids the server to pick up user
-            # files such as random.py or string.py instead of the
-            # standard library modules named the same.
-            if self.language == 'python':
-                cwd = get_conf_path('empty_cwd')
-                if not osp.exists(cwd):
-                    os.mkdir(cwd)
-            else:
-                cwd = None
+        # Set the PyLS current working to an empty dir inside
+        # our config one. This avoids the server to pick up user
+        # files such as random.py or string.py instead of the
+        # standard library modules named the same.
+        if self.language == 'python':
+            cwd = get_conf_path('empty_cwd')
+            if not osp.exists(cwd):
+                os.mkdir(cwd)
+        else:
+            cwd = None
 
-            self.lsp_server = subprocess.Popen(
-                self.server_args,
-                stdout=server_stdout,
-                stdin=server_stdin,
-                stderr=server_stderr,
-                creationflags=creation_flags,
-                cwd=cwd)
+        # Setup server
+        self.server = QProcess(self)
+        self.server.errorOccurred.connect(self.handle_process_errors)
+        self.server.setWorkingDirectory(cwd)
+        self.server.setProcessChannelMode(QProcess.MergedChannels)
+        if server_log_file is not None:
+            self.server.setStandardOutputFile(server_log_file)
```

Comment:
I think most servers write on stderr

Reply:
I set the channels mode a couple of lines above to `QProcess.MergedChannels`, which means stderr and stdout are merged into a single channel, as described [here](https://doc.qt.io/qt-5/qprocess.html#ProcessChannelMode-enum).